controller/embedding/manager.py

```diff
@@ -134,4 +104,110 @@ def get_embedding_name(
+
+    if model:
+        name += f"-{model}"
+
+    if apiToken:
+        name += f"-{apiToken[:8]}...{apiToken[-4:]}"
+    
+    return name
+
+
+def recreate_embeddings(project_id: str, embedding_ids: Optional[List[str]] = None) -> None:
+    if not embedding_ids:
+        embeddings = embedding.get_all_embeddings_by_project_id(project_id)
+        if len(embeddings) == 0:
+            return
+        embedding_ids = [str(embed.id) for embed in embeddings]
+    for embedding_id in embedding_ids:
+        embedding.update_embedding_state_waiting(project_id, embedding_id)
+        notification.send_organization_update(
+                project_id,
+                f"embedding:{embedding_id}:progress:{0.0}",
+            )
+        notification.send_organization_update(
+                project_id,
+                f"embedding:{embedding_id}:state:{enums.EmbeddingState.WAITING.value}",
+            )
+    general.commit()
+    for embedding_id in embedding_ids:
+        new_id = None
+        try:
+            embedding_item = embedding.get(project_id, embedding_id)
+            if not embedding_item:
+                continue
+            embedding_item = __recreate_embedding(project_id, embedding_id)
+            new_id = embedding_item.id
+            time.sleep(2)
+            while True:
+                embedding_item = general.refresh(embedding_item)
+                if not embedding_item:
+                    raise Exception("Embedding not found")
+                elif embedding_item.state == enums.EmbeddingState.FAILED.value or embedding_item.state == enums.EmbeddingState.FINISHED.value:
+                    break
+                else:
+                    time.sleep(1)
+        except Exception as e:
+            print(
+                f"Error while recreating embedding for {project_id} with id {embedding_id} - {e}", flush=True
+            )
+            notification.send_organization_update(
+                project_id,
+                f"embedding:{embedding_id}:state:{enums.EmbeddingState.FAILED.value}",
+            )
+            old_embedding_item = embedding.get(project_id, embedding_id)
+            if old_embedding_item:
+                old_embedding_item.state = enums.EmbeddingState.FAILED.value
+            
+            if new_id:
+                new_embedding_item = embedding.get(project_id, new_id)
+                if new_embedding_item:
+                    new_embedding_item.state = enums.EmbeddingState.FAILED.value
+            general.commit()
+
+
+        notification.send_organization_update(
+            project_id=project_id, message="embedding:finished:all"
+        )
+
+
+
+def __recreate_embedding(
+    project_id: str, embedding_id: str
+) -> Embedding:
+    old_embedding_item = embedding.get(project_id, embedding_id)
+    old_id = old_embedding_item.id
+    new_embedding_item = embedding.create(
+        project_id,
+        old_embedding_item.attribute_id,
+        old_embedding_item.name,
```

Comment:
this looks to be a copy of controller/embedding/util.py -> recreate_embedding. Didn't check every line so maybe I'm missing something but I think if possible we should reuse the recreate_embedding method

Reply:
This is indeed a new version of the one in util which is not used anymore but was not removed, i removed the one from util